public/app/features/dashboard-scene/edit-pane/DashboardEditPaneSplitter.tsx

```diff
@@ -0,0 +1,144 @@
+import { css, cx } from '@emotion/css';
+import React, { CSSProperties } from 'react';
+
+import { GrafanaTheme2 } from '@grafana/data';
+import { config, useChromeHeaderHeight } from '@grafana/runtime';
+import { useStyles2 } from '@grafana/ui';
+import NativeScrollbar from 'app/core/components/NativeScrollbar';
+
+import { useSnappingSplitter } from '../panel-edit/splitter/useSnappingSplitter';
+import { DashboardScene } from '../scene/DashboardScene';
+import { NavToolbarActions } from '../scene/NavToolbarActions';
+
+import { DashboardEditPaneRenderer } from './DashboardEditPane';
+
+interface Props {
+  dashboard: DashboardScene;
+  isEditing?: boolean;
+  body?: React.ReactNode;
+  controls?: React.ReactNode;
+}
+
+export function DashboardEditPaneSplitter({ dashboard, isEditing, body, controls }: Props) {
+  const headerHeight = useChromeHeaderHeight();
+  const styles = useStyles2(getStyles, headerHeight ?? 0);
+
+  if (!config.featureToggles.dashboardNewLayouts) {
+    return (
+      <NativeScrollbar onSetScrollRef={dashboard.onSetScrollRef}>
+        <div className={styles.canvasWrappperOld}>
+          <NavToolbarActions dashboard={dashboard} />
+          <div className={styles.controlsWrapperSticky}>{controls}</div>
+          <div className={styles.body}>{body}</div>
+        </div>
+      </NativeScrollbar>
+    );
+  }
+
+  const { containerProps, primaryProps, secondaryProps, splitterProps, splitterState, onToggleCollapse } =
+    useSnappingSplitter({
```

Comment:
I think this might break the hooks rules because we're using a hook after an `if`?

Reply:
yea, for some reason I am not getting any lint error for it. I was planing to do a es-lint ignore if it came up. (Since the feature toggle never changes during rendering it is fine)